### DISCUSSION TITLE
Progress Bar independent from pagination

### DISCUSSION THREAD

**Discussion by volysandro** [marp-team/marp#93] [Q&A]
Hello everyone!

So I am kind of stuck on a subject. I would like to create a progress bar in the footer, which is calculated directly from the pagination so:
data-marpit-pagination / data-marpit-pagination-total * 100 and then create a simple colored div from there.

How would I go about this, if I still want to keep the original pagination number (bottom right for me)?

I am not posting this because I want you to do it for me, I am posting it because I went really deep into the source and literally didn't find anything I could abuse for this 🤣 So thanks in advance for any help and pls be kind to me if this is an easy solve 🙈

**Comment by yhatt** [comment]
Marp CLI has already provided an easy solution for a progress bar within HTML presentation. Have you tried [`--bespoke.progress` option](https://github.com/marp-team/marp-cli#features)?

Any presentation contents including page number should work normally because this progress bar is working on _the outside of the context for Marp presentation_.

In the other hand, it's difficult to apply your custom style so might not be matched to your advanced wants, such as the progress bar in the footer (`--bespoke.progress` provides the bar into the top).

**Reply by volysandro** [reply]
Thanks yhatt! Unfortunately this is not exactly what I was searching for, since it would only render in html and the bar is not on the slides itself. I have found a way to do it yesterday and I have high hopes to share it with you all in due time :)
